test/common/http/http2/conn_pool_test.cc

```diff
@@ -1797,6 +1799,48 @@ TEST_F(Http2ConnPoolImplTest, TestStateWithMultiplexing) {
   CHECK_STATE(0 /*active*/, 0 /*pending*/, 0 /*capacity*/);
 }
 
+TEST_F(Http2ConnPoolImplTest, InitialStreams) {
```

Comment:
Nit: It would be nice to break this into N tests, each testing a single behavior of calculateInitialStreams(), and with a test name like that explains the behavior.